File: Design-Patterns/TemplateMethod/CSharp/Character.cs
namespace TemplateMethod
{
    public class Character
    {
        public Vector Position { get; set; }
        public int Health { get; protected set; }
        public int Mana { get; protected set; }

        public Character()
        {
            Position = new Vector(0, 0, 0);
            Health = 100;
            Mana = 100;
        }
    }
}

File: Design-Patterns/TemplateMethod/Cpp/character.cpp
#include "character.h"

character::character()
{
	health_ = 100;
	mana_ = 100;
	position = vector(0, 0, 0);
}

int character::get_health()
{
	return health_;
}

int character::get_mana()
{
	return mana_;
}
